recipes/paho-mqtt-c/all/conanfile.py

```diff
@@ -52,80 +49,89 @@ def config_options(self):
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["PAHO_ENABLE_TESTING"] = False
-        self._cmake.definitions["PAHO_BUILD_DOCUMENTATION"] = False
-        self._cmake.definitions["PAHO_ENABLE_CPACK"] = False
-        self._cmake.definitions["PAHO_BUILD_DEB_PACKAGE"] = False
-        self._cmake.definitions["PAHO_BUILD_ASYNC"] = self.options.asynchronous
-        self._cmake.definitions["PAHO_BUILD_STATIC"] = not self.options.shared
-        self._cmake.definitions["PAHO_BUILD_SHARED"] = self.options.shared
-        self._cmake.definitions["PAHO_BUILD_SAMPLES"] = False
-        self._cmake.definitions["PAHO_WITH_SSL"] = self.options.ssl
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PAHO_ENABLE_TESTING"] = False
+        tc.variables["PAHO_BUILD_DOCUMENTATION"] = False
+        tc.variables["PAHO_ENABLE_CPACK"] = False
+        tc.variables["PAHO_BUILD_DEB_PACKAGE"] = False
+        tc.variables["PAHO_BUILD_ASYNC"] = self.options.asynchronous
+        tc.variables["PAHO_BUILD_STATIC"] = not self.options.shared
+        tc.variables["PAHO_BUILD_SHARED"] = self.options.shared
+        tc.variables["PAHO_BUILD_SAMPLES"] = False
+        tc.variables["PAHO_WITH_SSL"] = self.options.ssl
         if self.options.ssl:
-            self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
-            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+            tc.cache_variables["OPENSSL_SEARCH_PATH"] = self.dependencies["openssl"].package_folder.replace("\\", "/")
+            tc.cache_variables["OPENSSL_ROOT_DIR"] = self.dependencies["openssl"].package_folder.replace("\\", "/")
         if self._has_high_performance_option:
-            self._cmake.definitions["PAHO_HIGH_PERFORMANCE"] = self.options.high_performance
-        self._cmake.configure()
-        return self._cmake
+            tc.variables["PAHO_HIGH_PERFORMANCE"] = self.options.high_performance
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_source(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                              "SET(CMAKE_MODULE_PATH \"${CMAKE_SOURCE_DIR}/cmake/modules\")",
-                              "LIST(APPEND CMAKE_MODULE_PATH \"${CMAKE_SOURCE_DIR}/cmake/modules\")")
+        apply_conandata_patches(self)
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
```

Comment:
Seems like it's needed with the new stuff https://github.com/conan-io/conan/blob/e0a8ee058bc8dc2d9811b8aeb6999f69aeb78d85/conan/tools/cmake/toolchain/blocks.py#L450

Reply:
@prince-chrismc 
Thanks you for lots of comments.

Last thing about CMAKE_MODULE_PATH, I think there are no way with `find_paths` to solve the issue original CMakeLists.txt sets CMAKE_MODULE PATH with single value.
Any advice would be appreciated.